### PR TITLE
Third party automation

### LINKED ISSUE #591: Automate Third Party Notices check/updates
Extra language may prove painful. We can potentially look to see if the extra language can be removed.

Would consider if this should be a per PR merge VS once a release action. If per PR, we could have it auto-submit a PR for updates so some human intervention can easily take place.

[after we remove cert bundle]

### PR DESCRIPTION
**Pull request by bizob2828**
## Proposed Release Notes
 * Added a pre-commit hook to check if package.json changes and run `oss third-party manifest` and `oss third-party notices`.  This will ensure the `third_party_manifest.json` and `THIRD_PARTY_NOTICES.md` up to date

## Links
Closes #591 

## Details
I will need to bump version of `@newrelic/newrelic-oss-cli` once I get these landed and new packages published:

https://github.com/newrelic/newrelic-oss-cli/pull/18
https://github.com/newrelic/newrelic-oss-cli/pull/17
https://github.com/newrelic/newrelic-oss-cli/pull/16

### CODE REVIEW

File: third_party_manifest.json [hunk 1]
```diff
@@ -367,6 +395,20 @@
       "licenseUrl": "https://github.com/sindresorhus/got/blob/v8.3.2/license",
       "licenseTextSource": "file"
     },
+    "http-errors@1.7.3": {
```
Comment (bizob2828) [inline third_party_manifest.json:398]: I'm goin to triage within the cli tool why this module is showing up twice

File: THIRD_PARTY_NOTICES_FOOTER.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+-----
+
+Portions copyright Node.js contributors. Depending on your existing libraries and package management settings,
+your systems may call externally maintained libraries in addition to those listed above.
+See [here](https://nodejs.org/en/docs/meta/topics/dependencies/) and [here](https://github.com/nodejs/node/blob/v4.3.1/LICENSE)
+for additional details regarding externally maintained libraries and certain related licenses and notices.”
```
Comment (carlo-808) [inline THIRD_PARTY_NOTICES_FOOTER.md:6]: I think we can remove the trailing double-quote here.

Comment (carlo-808) [inline THIRD_PARTY_NOTICES_FOOTER.md:6]:
```suggestion
for additional details regarding externally maintained libraries and certain related licenses and notices.
```

File: THIRD_PARTY_NOTICES.md [hunk 3]
```diff
@@ -3274,3 +3372,4 @@ Portions copyright Node.js contributors. Depending on your existing libraries an
 your systems may call externally maintained libraries in addition to those listed above.
 See [here](https://nodejs.org/en/docs/meta/topics/dependencies/) and [here](https://github.com/nodejs/node/blob/v4.3.1/LICENSE)
 for additional details regarding externally maintained libraries and certain related licenses and notices.”
```
Comment (carlo-808) [inline THIRD_PARTY_NOTICES.md:3374]:
```suggestion
for additional details regarding externally maintained libraries and certain related licenses and notices.
```

File: bin/update-third-party-notices.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/bin/sh
```
Comment (carlo-808) [inline bin/update-third-party-notices.sh:1]: Nice hook and script!

File: package.json [hunk 5]
```diff
@@ -137,6 +137,7 @@
     "ssl": "./bin/ssl.sh",
     "sub-install": "node test/bin/install_sub_deps",
     "test": "npm run integration && npm run unit",
+    "third-party-updates": "oss third-party manifest --includeOptDeps && oss third-party notices --includeOptDeps && git add THIRD_PARTY_NOTICES.md third_party_manifest.json",
```
Comment (carlo-808) [inline package.json:140]: 👍

File: THIRD_PARTY_NOTICES.md [hunk 6]
```diff
@@ -1792,23 +1669,10 @@ THE SOFTWARE.
...
-SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
-ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
-NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
-ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
-The views and conclusions contained in the software and documentation are those of the
-authors and should not be interpreted as representing official policies, either expressed
-or implied, of Gary Court or the JSON Schema specification.
-
+undefined
```
Comment (carlo-808) [inline THIRD_PARTY_NOTICES.md:1675]: I think we need to manually add this license in.

**Review by carlo-808 — COMMENTED** [review]
One license got missed when generating Third Party Notices.

**Comment by bizob2828** [comment]
Let's wait for https://github.com/newrelic/node-newrelic/pull/836 before we merge this as it'll fix the few things @carlo-808 called out in here around manifest and notice issues

**Review by michaelgoin — COMMENTED** [review]
Looks great. I appreciate pushing this to a pre-commit hook so these get updated with the relevant changes.

I am surprised the package-lock changed so much (6853 deletions) with the addition of the newrelic-oss-cli. Is there maybe some change from a different PR that accidentally ended up there?

Prob harmless but there was a test failure. I just kicked-off in case a flaky run but figured I'd mention. If is just from the added dev dep and things go green, I'll approve.

**Review by michaelgoin — APPROVED** [review]
👍

**Pull request merged**